src/node/rpc/member_frontend.h

```diff
@@ -400,6 +401,128 @@ namespace ccf
       return get_path_param(params, "member_id", member_id.value(), error);
     }
 
+    template <typename T>
+    void add_kv_wrapper_endpoint(T table)
+    {
+      constexpr bool is_map = nonstd::is_specialization<T, kv::TypedMap>::value;
+      constexpr bool is_value =
+        nonstd::is_specialization<T, kv::TypedValue>::value;
+
+      if constexpr (!(is_map || is_value))
+      {
+        static_assert(nonstd::dependent_false_v<T>, "Unsupported table type");
+      }
+
+      auto getter =
+        [&, table](endpoints::ReadOnlyEndpointContext& ctx, nlohmann::json&&) {
+          LOG_TRACE_FMT("Called getter for {}", table.get_name());
+          auto response_body = nlohmann::json::object();
+
+          auto handle = ctx.tx.template ro(table);
+          if constexpr (is_map)
+          {
+            handle->foreach([&response_body](const auto& k, const auto& v) {
+              if constexpr (
+                std::is_same_v<typename T::Key, ccf::CodeDigest> ||
+                std::is_same_v<typename T::Key, crypto::Sha256Hash>)
+              {
+                response_body[k.hex_str()] = v;
+              }
+              else
+              {
+                response_body[k] = v;
+              }
+              return true;
+            });
+          }
+          else if constexpr (is_value)
+          {
+            response_body = handle->get();
+          }
+
+          return ccf::make_success(response_body);
+        };
+
+      std::string_view table_name = table.get_name();
+      constexpr auto gov_prefix = "public:ccf.gov.";
+      constexpr auto internal_prefix = "public:ccf.internal.";
+      if (table_name.starts_with(gov_prefix))
+      {
+        table_name.remove_prefix(strlen(gov_prefix));
+      }
+      else if (table_name.starts_with(internal_prefix))
+      {
+        table_name.remove_prefix(strlen(internal_prefix));
+      }
+      else
+      {
+        throw std::logic_error(fmt::format(
+          "Should only be used to wrap governance tables. '{}' is not "
+          "supported",
+          table_name));
+      }
+
+      auto endpoint = make_read_only_endpoint(
+        fmt::format("/kv/{}", table_name),
```

Comment:
Because we've gone the governance-specialised rather than generic way, I think it'd be nice to replace the `.` with `/` in the table name so that it's more RESTful? Looking at https://microsoft.github.io/CCF/main/audit/builtin_maps.html, I can see that we've not been very consistent so far (e.g. `nodes.info` vs. `proposals_info`) but I think it'd be nice regardless?

Reply:
I'm on the fence about this, but for similar reasons as were listed in #4073 and then superceded - doing more mangling of the name makes it harder to do lookup (in either direction), and if these look like RESTy resources then they look like badly designed RESTy resources (`/gov/kv/nodes/*` vs `/node/network/nodes/*` - similar names, and overlapping content, how are you supposed to know which to call?).

But I think you're right - if we're going down this route we should go all the way.

Reply:
This is now done.